plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcDeletedRows.java

```diff
@@ -244,6 +251,10 @@ private RowId getRowId(int position)
                                 long row = BIGINT.getLong(page.getBlock(ROW_ID_INDEX), i);
                                 RowId rowId = new RowId(originalTransaction, bucket, statement, row);
                                 deletedRowsBuilder.add(rowId);
+                                rowCount++;
+                                if (rowCount % 1000 == 0) {
+                                    systemMemoryUsage.setBytes(rowCount * RowId.INSTANCE_SIZE);
```

Comment:
`sizeOfObjectArray(rowCount) + rowCount * RowId.INSTANCE_SIZE`

Reply:
Is it worth doing setting this as we loop vs just doing it once at the end?
Also, can we safely ignore the return value of `setBytes`?

Reply:
The benefit of doing that in the loop (I think) is that we can pause some other query for a while, allowing us to get to the end of loop without OOM. Other than that it does not change much I think.
@findepi am I missing something?

Reply:
we should assume that building `deletedRows` list can exhaust node memory.
if we set memory usage only at the end, we still do not protect ourselves against node OOM.

> Also, can we safely ignore the return value of setBytes?

we cannot block here (can we?).
we could use `trySetBytes` here, perhaps.